am(cors): domain level CORS doc (#1010)
gravitee-io/gravitee-docs: pages/am/3.x/user-guide/cors/user-guide-cors-configuration.adoc

= CORS Configuration
:page-sidebar: am_3_x_sidebar
:page-permalink: am/current/am_userguide_cors_configuration.html

> :page-folde![](../../../../../../../../../Desktop/graviteeio-am-userguide-cors-default.png)r: am/user-guide

Reviewer: ```suggestion
:page-folder: am/user-guide
```

NOTE: The default value for the origin is `*` and max age is `86400 seconds (1 day)` if you do not provide any values.

WARNING: Domain level CORS settings only affect the domain for which the settings are enabled.
You should use the settings in `gravitee.yml` file if you wish to use a single CORS configuration for all the domains.

== Disabling The Domain level CORS Configuration

Disabling the domain level CORS configuration enables gateway level CORS configuration automatically.
The gateway configuration uses values from the `gravitee.yml` file.
The default values are used for both domain level and gateway level configurations if custom values are unspecified.

== CORS Default Values

[cols="2",options="header"]
|=========================================================
|Property | Default Value

|allow-origin| *
|allow-methods| OPTIONS, GET, POST, PUT, DELETE, PATCH
|allow-headers| Cache-Control, Pragma, Origin, Authorization, Content-Type, X-Requested-With, If-Match, X-Xsrf-Token
|max-age| 864000 seconds (1 day)
|allow-credentials| false
|=========================================================

Here is the screenshot of these variables from the `gravitee.yml` file

> image::{% link images/am/current/graviteeio-am-userguide-cors-settings-yml.png %}[Gateway CORS defailt settings,500,400]

Reviewer: Why use a screenshot (which I find difficult to read) when you could just copy and paste an example directly from the `gravitee.yml` file?

== Brief Definition

=== Access-Control-Allow-Origin

Allow to specify one or multiples origin(s) that may access the resource.

> If you want to allow all origins, you can put `*` but it's not safe for production.

Reviewer: ```suggestion
If you want to allow all origins, you can put `*` but this is not safe for production.
```

=== Access-Control-Allow-Methods

Specifies the method or methods allowed when accessing the resource. This is used in response to a preflight request.

=== Access-Control-Allow-Headers

Here you specify the headers allowed to be used by your requests.
Typically, on your request headers you will have a header `Access-Control-Request-Headers` which request CORS to be configured to allow its values.

> Let's see a simple use case:

Reviewer: ```suggestion
==== Simple use case:
```
